testing/unittests/rtlxmltests.cpp

```diff
@@ -0,0 +1,415 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2025 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifdef _USE_CPPUNIT
+#include "rtlformat.hpp"
+#include "unittests.hpp"
+#include "eclrtl.hpp"
+
+#define CPPUNIT_ASSERT_EQUAL_STR(x, y) CPPUNIT_ASSERT_EQUAL(std::string(x ? x : ""), std::string(y ? y : ""))
+
+class RtlXmlTest : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE(RtlXmlTest);
+    CPPUNIT_TEST(testXmlString);
+    CPPUNIT_TEST(testXmlBool);
+    CPPUNIT_TEST(testXmlIntegers);
+    CPPUNIT_TEST(testXmlReal);
+    CPPUNIT_TEST(testXmlData);
+    CPPUNIT_TEST(testXmlDecimal);
+    CPPUNIT_TEST(testXmlUnicode);
+    CPPUNIT_TEST(testXmlUtf8);
+    CPPUNIT_TEST(testXmlNesting);
+    CPPUNIT_TEST(testXmlAttributes);
+    CPPUNIT_TEST(testJsonUnicode);
+    CPPUNIT_TEST(testJsonDecimal);
+    CPPUNIT_TEST(testComplexStructure);
+    CPPUNIT_TEST_SUITE_END();
+
+public:
+    void setUp() {}
+    void tearDown() {}
+
+    void testXmlString()
+    {
+        StringBuffer out;
+
+        // Test with fieldname
+        outputXmlString(5, "hello", "greeting", out);
+        CPPUNIT_ASSERT_EQUAL_STR("<greeting>hello</greeting>", out.str());
+        out.clear();
+
+        // Test with null fieldname
+        outputXmlString(5, "hello", nullptr, out);
+        CPPUNIT_ASSERT_EQUAL_STR("hello", out.str());
+        out.clear();
+
+        // Test with empty string fieldname
+        outputXmlString(5, "hello", "", out);
+        CPPUNIT_ASSERT_EQUAL_STR("hello", out.str());
+        out.clear();
```

Comment:
Is there a sense to add same empty string and nullptr test for 'field' parameter? (And both 'field' and 'fieldname')?